### PR TITLE
Add X-activestorage-count header test

### PR DESCRIPTION
**Pull request by sd109**
Also cleans up formatting using python black and fixes broken mocks

### CODE REVIEW

**Comment by sd109** [comment]
Full test suite passes when run against https://github.com/stackhpc/s3-active-storage-rs/pull/44 and also with test results mocked by setting `PROXY_URL = None` (Apologies, in hindsight I probably should have done the mocks fixing and formatting in different PRs...)

File: compliance/mocks.py [hunk 1]
```diff
@@ -13,6 +15,7 @@ def __init__(self, status_code: int, operation_result) -> None:
             'content-length': str(len(self.content)),
             'x-activestorage-dtype': str(operation_result.dtype),
             'x-activestorage-shape': str(list(operation_result.shape)),
+            'x-activestorage-count': str(array_data[tuple(slice(*s) for s in selection)].size if selection else array_data.size),
```
Comment (markgoddard) [inline compliance/mocks.py]: Doesn't the array data already have the selection applied here?

File: compliance/test_error_response.py [hunk 2]
```diff
@@ -73,7 +73,7 @@ def test_invalid_operation(monkeypatch):
 
     #Make proxy request
     if PROXY_URL is None:
-        monkeypatch.setattr(requests, 'post', lambda *args, **kwargs: MockResponse(status_code=404, operation_result=np.array([]))) #Is this a sensible mock response?
+        monkeypatch.setattr(requests, 'post', lambda *args, **kwargs: MockResponse(status_code=404, operation_result=np.array([])))
```
Comment (markgoddard) [inline compliance/test_error_response.py]: Not passing the request_data or array_data args here.

**Review by markgoddard — CHANGES_REQUESTED** [review]

File: compliance/mocks.py (hunk 1 above)
Comment (markgoddard) [reply compliance/mocks.py]: My bad, was just looking at the first commit...

File: compliance/test_error_response.py (hunk 2 above)
Comment (markgoddard) [reply compliance/test_error_response.py]: My bad, was just looking at the first commit...

**Review by markgoddard — APPROVED** [review]

**Comment by markgoddard** [comment]
Wait for corresponding change to merge.
